lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -454,11 +459,12 @@ TypeLoweringVisitor::getPreservatinoModeForModule(FModuleLike module) {
   if (!isa<FModuleOp>(module))
     return PreserveAggregate::None;
 
-  // If `module` is a top-module, we have to lower ports. Don't read attributes
-  // of `module` since the attributes could be mutated in a different thread.
+  // If preservePublicTypes is true, we have to lower ports of public modules.
```

Comment:
```suggestion
  // If preservePublicTypes is true, we have to preserve ports of public modules.
```
or "we can't lower the ports of public modules" or however.

Reply:
Oh, I see re: https://github.com/llvm/circt/pull/4654#discussion_r1105750516 that this is the intended behavior of `preservePublicTypes`.  :+1:.

Reply:
Yes, the option name is very confusing so I'll change 😅